src/transport.rs

```diff
@@ -27,7 +29,7 @@ const MESSAGE_QUEUE_SIZE: usize = 100;
 /// This socket handles the server-to-client half of the bidirectional communication stream.
 pub trait Loopback {
     /// Yields a stream of pending server-to-client requests.
-    type RequestStream: Stream<Item = Request>;
+    type RequestStream: Stream<Item = Request> + Unpin;
```

Comment:
Note: Breaking change, but it still pass the unit tests + works with stdio. Required by `stream_select!`